pkg/controller/instance/instance_controller.go

```diff
@@ -451,36 +448,12 @@ func startPlanExecution(i *v1beta1.Instance, planName string, ov *v1beta1.Operat
 		ensurePlanStatusInitialized(i, ov)
 	}
 
-	// update status of the instance to reflect the newly starting plan
-	notFound := true
-	for planIndex, v := range i.Status.PlanStatus {
-		if v.Name == planName {
-			// update plan status
-			notFound = false
-			planStatus := i.Status.PlanStatus[planIndex]
-			planStatus.Set(v1beta1.ExecutionPending)
-			planStatus.UID = uuid.NewUUID()
-			for j, p := range v.Phases {
-				planStatus.Phases[j].Set(v1beta1.ExecutionPending)
-				for k := range p.Steps {
-					i.Status.PlanStatus[planIndex].Phases[j].Steps[k].Set(v1beta1.ExecutionPending)
-				}
-			}
-
-			i.Status.PlanStatus[planIndex] = planStatus // we cannot modify item in map, we need to reassign here
-
-			// update activePlan and instance status
-			i.Status.AggregatedStatus.Status = v1beta1.ExecutionPending
-			i.Status.AggregatedStatus.ActivePlanName = planName
-
-			break
-		}
-	}
-	if notFound {
-		return &v1beta1.InstanceError{Err: fmt.Errorf("asked to execute a plan %s but no such plan found in instance %s/%s", planName, i.Namespace, i.Name), EventName: kudo.String("PlanNotFound")}
+	// reset newly starting plan status
```

Comment:
the rule right now is, a method becomes part of the `instance_type_helpers` once it is used in at least 2 different places. otherwise, there is no need to expose it.

Reply:
And even if it's used in two places: When we start with api versioning, i.e. `v1`, `v2beta1`, etc. we should really think about moving the `instance_type_helpers` out of there and into a package that is not in the API. Otherwise we need to duplicate that code with every new API version...

Reply:
> Otherwise we need to duplicate that code with every new API version...

which we might need to do, right?